src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO book(title,price) VALUE(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2,book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at least one row,"
+                        + " but inserted 0 rows");
+            }
+            ResultSet generateKeys = statement.getGeneratedKeys();
+            if (generateKeys.next()) {
+                long id = generateKeys.getObject(1,long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1,id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+                Book book = new Book();
+                book.setId(id);
+                book.setPrice(price);
+                book.setTitle(title);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a new connection to the DB", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM book";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                long id = resultSet.getInt("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                Book book = new Book();
+                book.setId(id);
+                book.setPrice(price);
+                book.setTitle(title);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a new connection to the DB", e);

```

Comment:
SQLException reason can be anything basically, not just the connection problem, so we can not be that specific in that case, but `e` will provide us wit everything we need to fix the problem in case we encounter it
```suggestion
            throw new DataProcessingException("Can't find all books", e);
```